File: c++/Math/Prime Factorization.cpp
Guarda en factors la lista de factores primos de n de menor a mayor. IMPORTANTE: Debe ejecutarse primero Sieve of Eratosthenes (al menos hasta un numero mayor a la raiz cuadrada de n).

vector<int> factors;

void primeFactors(int n) {
    factors.clear();
    for (int i = 0, p = primes[i]; p*p <= n; p = primes[++i]) {
        while (n % p == 0) {
            factors.push_back(p);
            n /= p;
        }
    }
    if (n > 1) factors.push_back(n);
}

File: java/Math/Prime Factorization.java
Guarda en primeFactors la lista de factores primos del value de menor a mayor. IMPORTANTE: Debe ejecutarse primero la criba de Eratostenes.  La criba debe existir al menos hasta la raiz cuadrada de value (se  recomienda dejar un poco de excedente).

import java.util.ArrayList;

static ArrayList<Long> primeFactors = new ArrayList<Long>();

static void calculatePrimeFactors(long value){
	primeFactors.clear();
	long temp = value;
	int factor;
	for (int i = 0; (long)primes.get(i) * primes.get(i) <= value; ++i){
		factor = primes.get(i);
		while (temp % factor == 0){
			primeFactors.add((long)factor);
			temp /= factor;
		}
	}
	if (temp != 1) primeFactors.add(temp);
}
